listenbrainz/background/background_tasks.py

```diff
@@ -0,0 +1,67 @@
+import time
+
+from flask import current_app
+from sqlalchemy import text
+
+from listenbrainz.background.delete import delete_listens_history, delete_user
+from listenbrainz.webserver import create_app, db_conn
+
+
+def add_task(user_id, task):
+    """ Add a task to the background tasks """
+    query = "INSERT INTO background_tasks (user_id, task) VALUES (:user_id, :task)"
+    db_conn.execute(text(query), {"user_id": user_id, "task": task})
+    db_conn.commit()
+
+
+def get_task():
+    """ Fetch one task from the database """
+    query = "SELECT * FROM background_tasks LIMIT 1"
+    result = db_conn.execute(text(query))
+    return result.first()
+
+
+def remove_task(task):
+    query = "DELETE FROM background_tasks WHERE id = :id"
+    db_conn.execute(text(query), {"id": task.id})
+    db_conn.commit()
+
+
+class BackgroundTasks:
+
+    def process_task(self, task) -> bool:
+        """ Perform the task and return whether the task succeeded """
+        if task.task == "delete_listens":
+            delete_listens_history(db_conn, task.user_id, task.created)
+        elif task.task == "delete_user":
+            delete_user(db_conn, task.user_id, task.created)
+        else:
+            current_app.logger.error(f"Unknown task type: {task}")
+        return True
+
+    def start(self):
+        current_app.logger.info("Background tasks processor started.")
+        while True:
+            try:
+                task = get_task()
+                current_app.logger.info(f"Processing task: {task}")
+                if task is None:
+                    current_app.logger.info("No pending tasks found, sleeping 5 seconds")
```

Comment:
remove. this will make the logs uselessly noisy.

Reply:
done